src/jit/compiler.h

```diff
@@ -2058,18 +2058,19 @@ class Compiler
 
 #if FEATURE_HW_INTRINSICS
     GenTreeHWIntrinsic* gtNewSimdHWIntrinsicNode(
-        var_types type, GenTree* op1, NamedIntrinsic hwIntrinsicID, var_types baseType, unsigned size);
+        var_types type, GenTree* op1, NamedIntrinsic hwIntrinsicID, var_types baseType, unsigned size, var_types srcType = TYP_UNDEF);
```

Comment:
Where is the new `srcType` field being consumed, and what is it needed for?

Reply:
Ah, I see, its under a different name, but consumed by the Conversion intrinsics.

Reply:
Yes, without it conversions cannot be encoded.

Reply:
This information should be encoded in a “overload” field to unify all the intrinsics whose codegen cannot be determined by intrinsicID + baseType rather than an additional field only for conversions.